### ISSUE TITLE
`toStrictEqual` should not compare return value of `get` function

### ISSUE THREAD

**Issue by yjhmelody** [jtenner/as-pect#344]
```
let a = 0;
class Foo {
    get foo(): i32 {
        return ++a;
    }
}

expect(new Foo()).toStrictEqual(new Foo());
```

```
  [Actual]: Foo {
      foo: 3
  }
[Expected]: Foo {
      foo: 4
  }
```
It not only compares, but also produces side effects.

**Comment by jtenner** [comment]
Yes. This behavior is mirrored in jest object comparisons. It was by design.

**Comment by yjhmelody** [comment]
> Yes. This behavior is mirrored in jest object comparisons. It was by design.

Can you provide a comparison without access to set/get ?

**Comment by jtenner** [comment]
Steps to prove this:

1. Install jest
2. create a test that uses `expect(someRef).toStrictEqual(anotherRef)`
3. validate the `get` property function on `someRef` is executed.

This behavior is by design.

If you think that you truly need strict equality, I might recommend wrapping your object with a special function like this:

```ts
function wrap(a: A): B {
  // copy each a value you want to test to a type B
  let b = new B();
  b.prop1 = a.prop1;
  return b;
}

expect(wrap(myA)).toStrictEqual(wrap(otherA));
```

**Comment by yjhmelody** [comment]
but this is not ts. how to define the function genericly？and compare the private field

**Comment by jtenner** [comment]
I can't deviate from developer expectations. Perhaps make a method on the class instead?

```ts
class Foo {
  __asBar(): Bar {
    // Make whatever you want to compare here
  }
}
```

Yes. This is not an elegant solution. Aspect happens to be very opinionated. I hope this answer suffices.

**Comment by jtenner** [comment]
> but this is not ts. how to define the function genericly？and compare the private field

By the way, private fields are compared. The problem is that computed fields are also compared. This is how jest currently works in production when working in a js environment.

**Comment by yjhmelody** [comment]
This is really bad, because the type deduction ability of As is much worse than that of TS. Many times a value val type is T | null, but it is not null at a certain position, so I need to cast val to T. The code for the cast is really Too much. So I wrote a get function to encapsulate the forced conversion logic, but even if it is private or not used in the unit test, this function will be called and then all panic. That is to say, the current design of as-pect affects the code implementation details of some as libraries.

**Comment by yjhmelody** [comment]
Well, I got it
